File: ex_happy-primes/submissions/thepandaatemyface-py/happyprime.py
l=n=input()
while l>4:l=sum(int(i)**2for i in str(l))
print['sad','happy'][l==1and str(n)!=1],
print['non-',''][n!=1 and sum(n%i==0for i in range(1,n))<2]+"prime"

File: ex_happy-primes/submissions/hobbs-pl/happyprime.pl
sub h{$_==1&& happy||$s{$_}++&& sad||do{$m=0;$m+=$_**2for split//;$_=$m;&h}}$n=$_=pop;die h,$",(1x$n)=~/^1?$|^(11+?)\1+$/&&"non-","prime\n"
